clippy_lints/src/ptr_offset_with_cast.rs

```diff
@@ -0,0 +1,130 @@
+use rustc::{declare_lint, hir, lint, lint_array, ty};
+use syntax::ast;
+use crate::utils;
+
+/// **What it does:** Checks for usage of the `offset` pointer method with a `usize` casted to an
+/// `isize`.
+///
+/// **Why is this bad?** If we’re always increasing the pointer address, we can avoid the numeric
+/// cast by using the `add` method instead.
+///
+/// **Known problems:** None
+///
+/// **Example:**
+/// ```rust
+/// let vec = vec![b'a', b'b', b'c'];
+/// let ptr = vec.as_ptr();
+/// let offset = 1_usize;
+///
+/// unsafe { ptr.offset(offset as isize); }
+/// ```
+///
+/// Could be written:
+///
+/// ```rust
+/// let vec = vec![b'a', b'b', b'c'];
+/// let ptr = vec.as_ptr();
+/// let offset = 1_usize;
+///
+/// unsafe { ptr.add(offset); }
+/// ```
+declare_clippy_lint! {
+    pub PTR_OFFSET_WITH_CAST,
+    style,
```

Comment:
Is this the right category?

Reply:
It would fit the complexity category. Feels less like a stylistic choice, more like in the scope of "use `x.is_empty()` instead of `x.len() == 0`.

Reply:
https://github.com/rust-lang-nursery/rust-clippy/pull/3104/commits/feb3e9fd5f8b24bcbe1a470b65dac66acb265721